test/manifoldc_test.cpp

```diff
@@ -19,29 +19,76 @@ TEST(CBIND, sphere) {
 
 TEST(CBIND, warp_translation) {
   size_t sz = manifold_manifold_size();
-  ManifoldVec3 (*warp)(float, float, float) = [](float x, float y, float z) {
+  ManifoldVec3 (*warp)(float, float, float, void *) = [](float x, float y,
+                                                         float z, void *) {
     ManifoldVec3 v = {x + 15.0f, y, z};
     return v;
   };
+  float *context = (float *)malloc(1 * sizeof(float));
+  context[0] = 15.0f;
+  ManifoldVec3 (*warpcontext)(
+      float, float, float, void *) = [](float x, float y, float z, void *ctx) {
+    ManifoldVec3 v = {x + ((float *)ctx)[0], y, z};
+    return v;
+  };
   ManifoldManifold *sphere = manifold_sphere(malloc(sz), 1.0f, 100);
   ManifoldManifold *trans = manifold_translate(malloc(sz), sphere, 15., 0., 0.);
-  ManifoldManifold *warped = manifold_warp(malloc(sz), sphere, warp);
+  ManifoldManifold *warped = manifold_warp(malloc(sz), sphere, warp, NULL);
```

Comment:
Can we test @pca006132's idea by simply omitting `NULL` here (and in the other tests)? If that works, then this is a pretty clean API. The only breaking change is removing `manifold_level_set_context`, which I'm comfortable with, especially since you're the main user of it.

Reply:
Unfortunately I can't get it to compile without the NULL in the tests.

I know there's tricks we can play with varargs or preprocessor but not sure how you would feel about that.

Reply:
No, that's fine, let's leave it. I assume C doesn't have nice things like default arguments? @pca006132 Did I misunderstand what you were saying?

Reply:
Yeah I mean something else. Will give an example later today.

Reply:
Okay, i guess we can skip this one. Is this ready to merge then?

Reply:
I'm happy if you're happy